### ISSUE TITLE
Fix stack_bands 

### ISSUE THREAD

**Issue by jhamman** [intake/intake-stac#55]
There have been some upstream changes in the eo extension that require us to rework the stack_bands method. I've marked the tests for this method upstream in #54.

Basically, the band info has moved to the asset properties:

```
  "assets": {
    ...
    "B1": {
      "href": "https://landsat-pds.s3.amazonaws.com/c1/L8/152/038/LC08_L1TP_152038_20200611_20200611_01_RT/LC08_L1TP_152038_20200611_20200611_01_RT_B1.TIF",
      "type": "image/tiff; application=geotiff",
      "eo:bands": [
        {
          "name": "B1",
          "common_name": "coastal",
          "center_wavelength": 0.44,
          "full_width_half_max": 0.02
        }
      ],
      "title": "Band 1 (coastal)"
    },
```

**Comment by jhamman** [comment]
@matthewhanson also shared a different pattern here:

https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l2a-cogs/
https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l2a-cogs/items

**Comment by scottyhq** [comment]
closed by #52
